### PR TITLE
Older GCC compatibility

### PR DESCRIPTION
**Pull request by glebm**
Now builds on GCC5

### CODE REVIEW

File: src/surface.cpp [hunk 1]
```diff
@@ -36,16 +36,16 @@ using namespace std;
...
+  return {
+      uint8_t(
+          compat::clamp(std::stoi(strColor.substr(0, 2), nullptr, 16), 0, 255)),
+      uint8_t(compat::clamp(
+          std::stoi(strColor.substr(2, 2).c_str(), nullptr, 16), 0, 255)),
+      uint8_t(compat::clamp(
+          std::stoi(strColor.substr(4, 2).c_str(), nullptr, 16), 0, 255)),
+      uint8_t(compat::clamp(
+          std::stoi(strColor.substr(6, 2).c_str(), nullptr, 16), 0, 255)),
+  };
```
Comment (pcercuei) [inline src/surface.cpp]: You changed tabs to spaces, please use the same code style.

**Review by pcercuei — COMMENTED** [review]
Please also add a small description to the commit. If it allows it to compile with GCC7, then just say that.

File: src/compat-filesystem.h [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#ifndef __COMPAT_FILESYSTEM__
+#define __COMPAT_FILESYSTEM__
+
+#if defined(__cplusplus) && __cplusplus >= 201703L && \
+    defined(__has_include) && __has_include(<filesystem>)
```
Comment (pcercuei) [inline src/compat-filesystem.h]: GCC doc seems to say that checking for __has_include and using it on the same line is not recommended:
https://gcc.gnu.org/onlinedocs/cpp/_005f_005fhas_005finclude.html

File: src/surface.cpp (hunk 1 above)
Comment (glebm) [reply src/surface.cpp]: I've added a `.clang-format` file in #2 to avoid accidents like this.

File: src/compat-filesystem.h (hunk 2 above)
Comment (glebm) [reply src/compat-filesystem.h]: Done

File: src/surface.cpp (hunk 1 above)
Comment (glebm) [reply src/surface.cpp]: Fixed

**Comment by glebm** [comment]
> Please also add a small description to the commit. If it allows it to compile with GCC7, then just say that.

Done for both commits

**Pull request merged**